From 856ffc19da1bd2b974dd6162c645746d6ff8b2b1 Mon Sep 17 00:00:00 2001
From: Andreas Dilger <adilger@whamcloud.com>
Date: Tue, 7 Jun 2011 10:22:29 -0600
Subject: [PATCH 6/6] mke2fs: Don't erase flash device if "-n" is given

Upstream commit: 8185ab9f38f0e9cd06feab9d8e59d059bde84bf6

If "mke2fs -n" is used, there should be no changes to the underlying
device.  Unfortunately, when the "discard" option was added in commit
c7cd908be59f48c66b4f3ac9a631ffe3dde4f1ab, it did not check for the "-n"
flag, and will discard all data on a flash device even if "-n" is given.

Check for the "noaction" flag before discarding any filesystem data.

Signed-off-by: Andreas Dilger <adilger@whamcloud.com>
Reviewed-by: Eric Sandeen <sandeen@redhat.com>
Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---
 misc/mke2fs.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/misc/mke2fs.c b/misc/mke2fs.c
index 0a027e5..772c250 100644
--- a/misc/mke2fs.c
+++ b/misc/mke2fs.c
@@ -2050,7 +2050,7 @@ int main (int argc, char *argv[])
 	}
 
 	/* Can't undo discard ... */
-	if (discard && (io_ptr != undo_io_manager)) {
+	if (!noaction && discard && (io_ptr != undo_io_manager)) {
 		retval = mke2fs_discard_device(fs);
 
 		if (!retval && io_channel_discard_zeroes_data(fs->io)) {
-- 
1.7.4.4

